suplemon/logger.py

```diff
@@ -50,14 +50,24 @@ def close(self):
 # Generate and configure handlers
 log_filepath = os.path.join(os.path.expanduser("~"), ".config", "suplemon", "output.log")
 logger_handlers = [
+    # Buffer 64k records in memory at a time
+    BufferingTargetHandler(64 * 1024, fd_target=sys.stderr),
+]
+
+# Error recovery when log_filepath isn't writable
+try:
+    if not os.path.exists(os.path.dirname(log_filepath)):
+        os.makedirs(os.path.dirname(log_filepath))
     # Output up to 4MB of records to `~/.config/suplemon/output.log` for live debugging
     # https://docs.python.org/3.3/library/logging.handlers.html#logging.handlers.RotatingFileHandler
     # DEV: We use append mode to prevent erasing out logs
     # DEV: We use 1 backup count since it won't truncate otherwise =/
-    RotatingFileHandler(log_filepath, mode="a", maxBytes=(4 * 1024 * 1024), backupCount=1),
-    # Buffer 64k records in memory at a time
-    BufferingTargetHandler(64 * 1024, fd_target=sys.stderr),
-]
+    rfh = RotatingFileHandler(log_filepath, mode="a+", maxBytes=(4 * 1024 * 1024), backupCount=1)
+    logger_handlers.append(rfh)
+except:
```

Comment:
Catching a generic exception is probably a bad idea. Maybe we should use `except OSError` and check `err.errno` for `EPERM` or similar?

https://docs.python.org/2/library/exceptions.html#exceptions.OSError

https://docs.python.org/2/library/errno.html

```python
except OSError as err:
  # If it was a permissioning error, log to stderr about it
  if err.errno == errno.EPERM
    print("Unable to open ~/.config/suplemon/output.log", file=sys.stderr)
  # Otherwise, raise the error
  else:
    raise
```